### DISCUSSION TITLE
Fully self-hosted

### DISCUSSION THREAD

**Discussion by sergi0g** [outline/outline#5106] [Self Hosting]
Hi, it's me again. I think I need to sum up what I want to do in 1 question.

My goal is to be able to host outline on my server for the following reasons
- I want full control over my data
- I want it to be accessible when offline, just by connecting to my server on the local network
- I want everything on my own server (minio, postgresql, etc)
- Self hosting is awesome

So I want to have a full outline instance on my server. I have tried many configs, but I can't seem to do it no matter which docs I follow. Do I need to have a domain name, or can I just use my private ip? What configuration does Keycloak (or Gitea, I don't mind) need, so I can sign in?

So I am basically asking for a full guide for my use case, since I can't find anything working.

**Comment by k1ck3r** [comment]
I hope this is some kind of a troll :/
Anyhow, if you go through documentation published here, do some google search you'll be able to find way too many HOWTOs. 

Everything above in your "question" or "request" is even easily deployable if you go through all of docs (pay attention at minio's installation and really look into their technical specifications - almost all of the minio's issues are based on NOT reading:) 

After all, i don't want to be rude, so i'll just give you a link, which i literally spent 30s on Google: https://blog.gurucomputing.com.au/s/blog/doc/installing-outline-tXZO9ehoFV

Again, Read. Carefully

**Reply by Limezy** [reply]
You can also install using Yunohost https://install-app.yunohost.org/?app=outline

**Reply by sergi0g** [reply]
> I hope this is some kind of a troll :/
> Anyhow, if you go through documentation published here, do some google search you'll be able to find way too many HOWTOs. 
> 
> Everything above in your "question" or "request" is even easily deployable if you go through all of docs (pay attention at minio's installation and really look into their technical specifications - almost all of the minio's issues are based on NOT reading:) 
> 
> After all, i don't want to be rude, so i'll just give you a link, which i literally spent 30s on Google: https://blog.gurucomputing.com.au/s/blog/doc/installing-outline-tXZO9ehoFV
> 
> Again, Read. Carefully

I had mentioned that I do not want to use a domain for this and use my private IP. I want a server that is capable of running offline. This is what all those guides lack. Everyone just assumes you own a domain. Well I don't plan to.

**Reply by tommoor** [reply]
I don’t think this is possible, as outline assumes SSL and you cannot have an SSL certificate for a private IP address

**Reply by sergi0g** [reply]
> I don’t think this is possible, as outline assumes SSL and you cannot have an SSL certificate for a private IP address

How can I make it not need SSL?

**Reply by Limezy** [reply]
I suggest you first start settling a dev environnement on your computer. After all it's basically running Outline offline without SSL.

**Reply by sergi0g** [reply]
> I suggest you first start settling a dev environnement on your computer. After all it's basically running Outline offline without SSL. 

Is that possible in a docker container?

**Reply by Limezy** [reply]
https://docs.getoutline.com/s/hosting/doc/local-development-5hEhFRXow7

**Reply by sergi0g** [reply]
Thanks!

**Reply by vgwidt** [reply]
You could just use a local domain and generate a cert yourself with OpenSSL.  That way you can still just follow the guides.

Otherwise you can use non-SSL with Docker by making sure your start command in docker-compose is includes `--env production-ssl-disabled` which would be the same effect as running it in development mode.  For example:
```
command: sh -c "yarn sequelize:migrate --env production-ssl-disabled && yarn start"
```

**Reply by Limezy** [reply]
Obviously, none of these is recommended for production usage

**Comment by sergi0g** [comment]
I managed to get outline to run on a github codespace with [https://github.com/vicalloy/outline-docker-compose](this project)

I will try on my local server as well.

**Reply by Altagen** [reply]
@hex-developer 

Hello everyone,
I did a medium article to get outline running on localhost, fully self hosted.

https://link.medium.com/yhtWcqQxNyb

Have a good day,
Ange

**Reply by MetaBliss** [reply]
> Hello everyone, I did a medium article to get outline running on localhost, fully self hosted.
> 
> https://link.medium.com/yhtWcqQxNyb
> 
> Have a good day, Ange

This is fantastic work. Quick question -- I'm hosting Outline with SSL (with my own domain name) so its not quite the same as your set up which is just local. However, I had troubles getting images to upload or to upload a team logo. It seems to be a problem with the Minio buckets unfortunately. I was wondering if you came across how to resolve this problem with Minio bucket policy?

**Reply by Altagen** [reply]
> > Hello everyone, I did a medium article to get outline running on localhost, fully self hosted.
> > https://link.medium.com/yhtWcqQxNyb
> > Have a good day, Ange
> 
> This is fantastic work. Quick question -- I'm hosting Outline with SSL (with my own domain name) so its not quite the same as your set up which is just local. However, I had troubles getting images to upload or to upload a team logo. It seems to be a problem with the Minio buckets unfortunately. I was wondering if you came across how to resolve this problem with Minio bucket policy?

Hey @MetaBliss , 
Thank you for your answer.

Yes, indeed all that concerns the storage of images, files etc corresponds to the minio. To make the minio work correctly, there are several solutions, but the simplest remains the one I explained in the minio part of my article. 

Can you explain to me more precisely what your configuration is?
Having a domain name does not change the procedure, but rather the links that will be the problem.

Indeed, I gave the names of my service "localhost" for keycloak and outline, since they are the two who must communicate between them. If you manage to make sure that no matter the configuration, the fact of returning to your domain name such as outline.yourdomainname.com, returns you each time to outline, and the same for keycloak.yourdomainename.com you should be able to get it working the exact same thing.

However, if your minio doesn't work (which is "storage" in the docker compose file) it means that you probably didn't take the corresponding files where I describe the env variables. 
Be careful that your docker composes can still have redirects to your DNS

Let me know your answer so I can try to provide better answers,
Ange

**Reply by sergi0g** [reply]
> @hex-developer 
> 
> Hello everyone,
> I did a medium article to get outline running on localhost, fully self hosted.
> 
> https://link.medium.com/yhtWcqQxNyb
> 
> Have a good day,
> Ange

That's great! I'll try it out and let you know!

**Comment by LukasMueller187** [comment]
I can somehow comprehend the authors opinion / requests. Even though it is mentioned that his requests are available, I myself was not able to deploy a solely local instance of outline unfortunately.
I was delighted at first to see a docker container available and thought "wohoo a quick deploy for testing" but man was I wrong. After spending a half day reading into whatever minio is and struggling with the ssl enforcing I sadly gave up. (I didn't even get to the point with auth).

Just wanted to share my experience, as a not too newbish dev who does run several apps / servers (but without AWS e. g.) I wasn't able to get a quick testing version for outline up and running. (btw. SSL isn't the problem but I don't want to request names and certificates for an application I just want to try if it fits our needs).

The app, however, does look stunning and I'm really sad I can't test it / use it. Hosted variant on the cloud isn't an option. I'd appreciate an easier deploying method :-)

**Reply by Altagen** [reply]
Please look at other threads, I already answered with an article especially for this purpose.

https://link.medium.com/9Yo1S9ShIzb

Have fun deploying Outline 
Regards, 
Ange

**Reply by LukasMueller187** [reply]
Thanks for the response. I already did see your posting but skipped it because of it's length.
I retried and got it to work (in a few hours), however for my testing purposes I did change some stuff like e. g. one docker-compose including the keycloak stuff.
One thing: maybe you should state the `FORCE_HTTPS=false` setting somewhere in the text.

Still, getting a local test version up and running is still too cumbersome in my opinion.

But guys, the application itself is stunning and lovely clean, really did enjoy using it so far :heart:. Will consider deploying on a full setup (with dns etc.).

**Reply by Altagen** [reply]
Happy it worked well. 
Yes it's long, However i needed it to be exhaustive. I tried my Best to explain everything as precisely as possible in order to be deployed by everyone. 

I'll do the change you advised, However I never got this error. Could you provide the version and the name of your browser please ? 

Have fun using outline,
Regards,
Ange

**Reply by LukasMueller187** [reply]
I was simply redirected to https all the time while trying to access the application, while `FORCE_HTTPS=true`. At least in firefox.

**Reply by Altagen** [reply]
Maybe it's your Firefox configuration that apply more security and always redirect to https. Using Firefox, I had to force Firefox not redirecting when I did the tutorial.

I added the line in medium, thank you for your feed back.

Regards,
Ange

**Reply by mfernrdx** [reply]
@Ange-Cesari 

Is it possible for you to repost the Medium article. The link provided no longer works. 

Best,
Michael

**Reply by Altagen** [reply]
Hello @mfernrdx 

Here is the link, sorry for the delay.

https://medium.com/@ange.cesari/how-to-deploy-outline-wiki-fully-self-hosted-no-tls-with-docker-compose-e50b57baaf73

Best,
Ange

**Reply by Altagen** [reply]
Hey @ShikharPOneHash
Could you please tell me where can you read the "make up" command in my tutorial was it on a screenshot, a command ?

Could you please provide more information ? 
Thank you

Best,
Ange

**Reply by Altagen** [reply]
Once everything is setup (compose + .env)

You have to start your keycloak first (with docker compose up) and setup everything for the client (Just as in the article) 

Then, same for your outline docker compose.

Once everything is set, you should be able to access your minio. You create the bucket and retrieve the keys. Then you docker compose down the outline docker compose (don't shut the keycloak) .

Once it's done, you fill up all the env variables that concern minio and keycloak in the .env as it's written in the article.

Then you start the outline docker compose, when everything is set correctly you don't need any command, following the article precisely is all you need.

**Reply by Altagen** [reply]
The command with yarn should be, as written in the article, run the first time with the docker compose of outline, when you're about to start the minio.

In the docker compose, i did the whole package with minio and outline. If you did a minio instance outside of this docker compose you need to remind that the first time ever you boot up the docker compose with docker compose up, you'll set up everything because you will need afterward to down you compose and up it once again.

This command is done, as written in the article once again, to fill the scheme in the database, so at the first start of outline, the postgres is filled with necessary information, and at the second start, outline will automaticaly be provided with the good informations in the database. That's used to setup the outline.

For the rest it seems right but you never described the error you get from this "crash". I cannot help you if you Just tell me what I wrote in the article, step by step.

Regards,
Ange

**Reply by Altagen** [reply]
First problem, I see that there is only "Running 1/1" means in your docker-compose.yml there is only one service.

My question is : Where are the redis, the postgres and the minio ? I have the impression that you read my article and you changed things in the configuration. Please, paste me your outline docker-compose.yml configuration you have right now.

If you did things correctly as I did in my article you should have 1 keycloak with 1 postgres in one docker compose, and 1 postgres, 1 redis, 1 minio and 1 outline service on the other docker compose. 

Also, there is "1/1" Running which means that only one service is started, which should not be the case. If you didn't change anything It means that you didn't do the "docker compose down" as its written in the article, between the database migration and the restart, and the outline has been stopped after the database migration and only it restarted.

Once the database migration is done, please do not REUSE the same command, you should just use `docker compose down` then `docker compose up`

**Comment by jenny787** [comment]
i have tried the setup explained in the medium article.  I get the same docker compose file as the user above (and I replaced the image with quay).  The minio setup worked fine, the keycloak setup worked fine, but at the last step when i try to access the outline site at:
http://ip:3000
It just redirects to https with an unable to connect message.  Any help? should I post my compose and env files?

**Reply by Altagen** [reply]
Hey jenny please do so

**Reply by jenny787** [reply]
docker compose
```
version: "3"
services:

  outline:
    image: docker.io/outlinewiki/outline
    env_file: ./docker.env
    ports:
      - "3000:3000"
    depends_on:
      - postgres
      - redis
      - storage
    networks:
      - keycloak_stack

  redis:
    image: docker.io/redis
    env_file: ./docker.env
    ports:
      - "6379:6379"
    volumes:
      - ./redis.conf:/redis.conf
    command: ["redis-server", "/redis.conf"]
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 10s
      timeout: 30s
      retries: 3
    networks:
      - keycloak_stack



  postgres:
    image: docker.io/postgres
    env_file: ./docker.env
    user: postgres
    ports:
      - "5432:5432"
    volumes:
      - database-data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD", "pg_isready"]
      interval: 30s
      timeout: 20s
      retries: 3
    environment:
      POSTGRES_USER: 'postgres'
      POSTGRES_PASSWORD: 'pass'
      POSTGRES_DB: 'outline'
    networks:
      - keycloak_stack

  storage:
    image: quay.io/minio/minio
    env_file: ./docker.env
    ports:
      - "9000:9000"
      - "9001:9001"
    entrypoint: sh
    command: -c 'minio server --console-address ":9001" /data/minio'
    deploy:
      restart_policy:
        condition: on-failure
    volumes:
      - storage-data:/data
    environment:
      MINIO_ROOT_USER: 'minio'
      MINIO_ROOT_PASSWORD: 'miniostorage'
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3
    networks:
      - keycloak_stack


volumes:
  storage-data:
  database-data:

networks:
  keycloak_stack:
    external: true
```

docker env
```
# –––––––––––––––– REQUIRED ––––––––––––––––

NODE_ENV=production

# Generate a hex-encoded 32-byte random key. You should use `openssl rand -hex 32`
# in your terminal to generate a random value.
SECRET_KEY=07d9aad2abb991bac0e462475af9827a0b38b985072d4dec603d5be651d361e7

# Generate a unique random key. The format is not important but you could still use
# `openssl rand -hex 32` in your terminal to produce this.
UTILS_SECRET=4cfc38a3b17dfb2b4384930a27c84fb9c72197c2af22bb2666a8883cb9c391cf
# For production point these at your databases, in development the default
# should work out of the box.
DATABASE_URL=postgres://postgres:pass@postgres:5432/outline
DATABASE_URL_TEST=postgres://postgres:pass@postgres:5432/outline-test
DATABASE_CONNECTION_POOL_MIN=
DATABASE_CONNECTION_POOL_MAX=
# Uncomment this to disable SSL for connecting to Postgres
PGSSLMODE=disable

# For redis you can either specify an ioredis compatible url like this
REDIS_URL=redis://redis:6379
# or alternatively, if you would like to provide additional connection options,
# use a base64 encoded JSON connection option object. Refer to the ioredis documentation
# for a list of available options.
# Example: Use Redis Sentinel for high availability
# {"sentinels":[{"host":"sentinel-0","port":26379},{"host":"sentinel-1","port":26379}],"name":"mymaster"}
# REDIS_URL=ioredis://eyJzZW50aW5lbHMiOlt7Imhvc3QiOiJzZW50aW5lbC0wIiwicG9ydCI6MjYzNzl9LHsiaG9zdCI6InNlbnRpbmVsLTEiLCJwb3J0IjoyNjM3OX1dLCJuYW1lIjoibXltYXN0ZXIifQ==

# URL should point to the fully qualified, publicly accessible URL. If using a
# proxy the port in URL and PORT may be different.
URL=http://outline:3000
PORT=3000


# See [documentation](docs/SERVICES.md) on running a separate collaboration
# server, for normal operation this does not need to be set.
COLLABORATION_URL=

# To support uploading of images for avatars and document attachments an
# s3-compatible storage must be provided. AWS S3 is recommended for redundancy
# however if you want to keep all file storage local an alternative such as
# minio (https://github.com/minio/minio) can be used.

# A more detailed guide on setting up S3 is available here:
# => https://wiki.generaloutline.com/share/125de1cc-9ff6-424b-8415-0d58c809a40f
#
AWS_ACCESS_KEY_ID=GYCe1ICvJ0s0OYoRxcr0
AWS_SECRET_ACCESS_KEY=Zepj7I16w2htvre0LNjhroi0Qjuc2JBCoqANNHmG
AWS_REGION=local
AWS_S3_ACCELERATE_URL=
AWS_S3_UPLOAD_BUCKET_URL=http://storage:9000
AWS_S3_UPLOAD_BUCKET_NAME=outline-bucket
AWS_S3_UPLOAD_MAX_SIZE=26214400
AWS_S3_FORCE_PATH_STYLE=true
AWS_S3_ACL=private


# –––––––––––––– AUTHENTICATION ––––––––––––––

# Third party signin credentials, at least ONE OF EITHER Google, Slack,
# or Microsoft is required for a working installation or you'll have no sign-in
# options.

# To configure Slack auth, you'll need to create an Application at
# => https://api.slack.com/apps
#
# When configuring the Client ID, add a redirect URL under "OAuth & Permissions":
# https://<URL>/auth/slack.callback
SLACK_CLIENT_ID=get_a_key_from_slack
SLACK_CLIENT_SECRET=get_the_secret_of_above_key

# To configure Google auth, you'll need to create an OAuth Client ID at
# => https://console.cloud.google.com/apis/credentials
#
# When configuring the Client ID, add an Authorized redirect URI:
# https://<URL>/auth/google.callback
GOOGLE_CLIENT_ID=
GOOGLE_CLIENT_SECRET=

# To configure Microsoft/Azure auth, you'll need to create an OAuth Client. See
# the guide for details on setting up your Azure App:
# => https://wiki.generaloutline.com/share/dfa77e56-d4d2-4b51-8ff8-84ea6608faa4
AZURE_CLIENT_ID=
AZURE_CLIENT_SECRET=
AZURE_RESOURCE_APP_ID=

# To configure generic OIDC auth, you'll need some kind of identity provider.
# See documentation for whichever IdP you use to acquire the following info:
# Redirect URI is https://<URL>/auth/oidc.callback
OIDC_CLIENT_ID=outline-client
OIDC_CLIENT_SECRET=O7o1zUS58QG7DEYwI90tB0pzh6PLALUr
OIDC_AUTH_URI=http://keycloak:8080/realms/outline/protocol/openid-connect/auth
OIDC_TOKEN_URI=http://keycloak:8080/realms/outline/protocol/openid-connect/token
OIDC_USERINFO_URI=http://keycloak:8080/realms/outline/protocol/openid-connect/userinfo

# Specify which claims to derive user information from
# Supports any valid JSON path with the JWT payload
OIDC_USERNAME_CLAIM=preferred_username

# Display name for OIDC authentication
OIDC_DISPLAY_NAME=Keycloak

# Space separated auth scopes.
OIDC_SCOPES=openid profile email


# –––––––––––––––– OPTIONAL ––––––––––––––––

# Base64 encoded private key and certificate for HTTPS termination. This is only
# required if you do not use an external reverse proxy. See documentation:
# https://wiki.generaloutline.com/share/1c922644-40d8-41fe-98f9-df2b67239d45
SSL_KEY=
SSL_CERT=

# If using a Cloudfront/Cloudflare distribution or similar it can be set below.
# This will cause paths to javascript, stylesheets, and images to be updated to
# the hostname defined in CDN_URL. In your CDN configuration the origin server
# should be set to the same as URL.
CDN_URL=

# Auto-redirect to https in production. The default is true but you may set to
# false if you can be sure that SSL is terminated at an external loadbalancer.
FORCE_HTTPS=false

# Have the installation check for updates by sending anonymized statistics to
# the maintainers
ENABLE_UPDATES=true

# How many processes should be spawned. As a reasonable rule divide your servers
# available memory by 512 for a rough estimate
WEB_CONCURRENCY=1

# Override the maximum size of document imports, could be required if you have
# especially large Word documents with embedded imagery
MAXIMUM_IMPORT_SIZE=5120000

# You can remove this line if your reverse proxy already logs incoming http
# requests and this ends up being duplicative
DEBUG=cache,presenters,events,emails,mailer,utils,multiplayer,server,services

# Configure lowest severity level for server logs. Should be one of
# error, warn, info, http, verbose, debug and silly
LOG_LEVEL=debug

# For a complete Slack integration with search and posting to channels the
# following configs are also needed, some more details
# => https://wiki.generaloutline.com/share/be25efd1-b3ef-4450-b8e5-c4a4fc11e02a
#
SLACK_VERIFICATION_TOKEN=your_tokenc
SLACK_APP_ID=A0XXXXXXX
SLACK_MESSAGE_ACTIONS=true

# Optionally enable google analytics to track pageviews in the knowledge base
GOOGLE_ANALYTICS_ID=

# Optionally enable Sentry (sentry.io) to track errors and performance,
# and optionally add a Sentry proxy tunnel for bypassing ad blockers in the UI:
# https://docs.sentry.io/platforms/javascript/troubleshooting/#using-the-tunnel-option)
SENTRY_DSN=
SENTRY_TUNNEL=

# To support sending outgoing transactional emails such as "document updated" or
# "you've been invited" you'll need to provide authentication for an SMTP server
SMTP_HOST=
SMTP_PORT=
SMTP_USERNAME=
SMTP_PASSWORD=
SMTP_FROM_EMAIL=
SMTP_REPLY_EMAIL=
SMTP_TLS_CIPHERS=
SMTP_SECURE=true

# The default interface language. See translate.getoutline.com for a list of
# available language codes and their rough percentage translated.
DEFAULT_LANGUAGE=en_US

# Optionally enable rate limiter at application web server
RATE_LIMITER_ENABLED=true

# Configure default throttling parameters for rate limiter
RATE_LIMITER_REQUESTS=1000
RATE_LIMITER_DURATION_WINDOW=60

FORCE_HTTPS=false
```

**Reply by Altagen** [reply]
have you set up the hosts file, so when you type "http://outline:3000" it redirects you to outline WebUI ?

Also, if the problem is a https redirection, try changing your web browser, what ever you are using, or check the security settings that may prevent you from accessing http only website.

Setting the HTTPS_FORCE=false should have already solve the problem and I can't think of anything else right now.

The only issue you could have is the /etc/host file on Linux. Please adapt for Windows if it's your platform.

Also, if you still have access trouble, please open the developer console tool and try debugging so you can provide more hints, for example request status code error.

**Reply by jenny787** [reply]
oh i see. I might have a problem. It's running on a linux machine but on a windows ad larger network.  I think i need to use the production version.  can i use the same keycloak container here with the production compose?

**Reply by Altagen** [reply]
I can't give you any advice for production purpose as I only support local version. All the variables for security are deactivated, ports in the container are exposed... According to me it's not a really good idea but you are free to try.

If i were you, for test purposes, I would deploy a Linux in a distant VM, Setup everything as you did until now, and access remotely the vm without using Windows.

**Reply by jenny787** [reply]
I realize my error.  I did not add it to my reverse proxy.  I just added an entry and i was able to ge to the login page.  But i need to change the outline domain to something else....like outline.domain1.com .?

**Reply by Altagen** [reply]
Well, if you read what I wrote in the article, docker automatically redirects to the correct container, using the container name, as an IP address.

For example, let's say your outline container is situated in 172.10.10.3, having the container name outline allows docker to forward the request to the container. But it only works for container that are in the same network.

That is why you have the keycloak_stack network in the outline docker compose, and  the "stack" network in your keycloak compose. Because having "stack" in keycloak creates automatically another network that is linked to stack, because it's external. So in order to have outline on the same network as keycloak, you have to attach to outline the keycloak network (now keycloak_stack) .

It avoids doing one docker compose that is hard to manage to do the required manipulations (create keycloak, setup the client id, Setup minio etc.)

The only difficulty it adds is the down order.

Now the benefits are that in the outline variables, you can have the expose name as "outline:3000" because the containers knows that docker redirects outline:3000 to the container and its port.

On the other hand, for the client ID, as you don't have the container's ip you are obliged to use the domaine name. But because the only domaine name you can find is outline because you are in the same network (with keycloak stack and stack). So in order for your browser to forward the request, you need to modify the etc/hosts file so that localhost is linked to outline and keycloak. By doing so, you "assimilate" the outline domain name with the outline docker container name and you are able to forward the keycloak request.

If you are in production environment, do not forget that using a reverse proxy allows to forward to the correct machine, but it doesn't mean that the communication between keycloak and outline is guarantee.

**Comment by heyvaldemar** [comment]
hey guys, I wrote [a detailed guide](https://www.heyvaldemar.com/install-outline-and-keycloak-using-docker-compose/) after I was able to configure Outline with MinIO and Keycloak.

Hopefully, it will help others.

**Reply by weironz** [reply]
> 大家好，在我能够使用 MinIO 和 Keycloak 配置 Outline 后，我写了[一篇详细的指南](https://www.heyvaldemar.com/install-outline-and-keycloak-using-docker-compose/)。
> 
> 希望它能帮助其他人。

Thanks, this has helped me a lot and I've put together a detailed step-by-step installation process as well：
https://blog.csdn.net/networken/article/details/141889199
architecture
![image](https://github.com/user-attachments/assets/86f540f6-fc48-4ff3-b258-c8d59512ee64)

**Reply by zung102** [reply]
I have set up self-hosted Outline successfully. Now I want to migrate documents from Onenote over but could not find a way to do so. Anyone has done this and can share the process? I found on the web to use 'import' function but it does not support Onenote export file.
